cmd/vcluster/context/flag_options.go

```diff
@@ -159,7 +164,8 @@ func AddFlags(flags *pflag.FlagSet, options *VirtualClusterOptions) {
 	flags.StringVar(&options.HostMetricsBindAddress, "host-metrics-bind-address", "0", "If set, metrics for the controller manager for the resources managed in the host cluster will be exposed at this address")
 	flags.StringVar(&options.VirtualMetricsBindAddress, "virtual-metrics-bind-address", "0", "If set, metrics for the controller manager for the resources managed in the virtual cluster will be exposed at this address")
 
-	flags.BoolVar(&options.RewriteHostPaths, "rewrite-host-paths", false, "If enabled, syncer will rewite hostpaths in synced pod volumes")
+	flags.BoolVar(&options.MountPhysicalHostPaths, "mount-physical-host-paths", false, "If enabled, syncer will rewite hostpaths in synced pod volumes")
+	flags.BoolVar(&options.EnsureMountPropagation, "ensure-mount-propagation", true, "If enabled, syncer will check certain hostpaths in synced pod volumes for mount propagation and enable correctly")
```

Comment:
@FabianKramm the default value here is true which means this will always check mount propagation and rewrite host paths to `/tmp/...` for select containers. However it does not effect because without the hpm component (wether central or daemon) its not gonna work anyway. Also if we were to not do this rewrite its not going to deviate in terms of behavior. What it enables us to do is - no config is needed when central HPM comes into picture.

Reply:
Also we can get rid of this flag entirely, but I choose to keep it in case someone wants to explicitly turn off hostpath rewrite and mount propagation even for those select containers